osu.Game/Screens/Play/Player.cs

```diff
@@ -849,9 +843,17 @@ private bool onFail()
             return true;
         }
 
-        // Called back when the transform finishes
+        /// <summary>
+        /// Invoked when the fail animation has finished.
+        /// </summary>
         private void onFailComplete()
         {
+            // fail completion is a good point to mark a score as failed,
+            // since the last judgement that caused the fail only applies to score processor after onFail.
+            // todo: this should probably be handled better.
+            Score.ScoreInfo.Passed = false;
```

Comment:
Why couldn't this be moved inside `FailScore()`?

Reply:
I've kept this out since it's not part of `ScoreProcessor`'s responsibility (while rank is, since it maps to the processor's `Rank` bindable), can move inside if you think otherwise.

Reply:
Please move inside then.